src/Modules/LiveRankingModule/Services/LiveRankingService.cs

```diff
@@ -242,15 +242,18 @@ private LiveRankingWidgetPosition RankingToTime(ExpandedLiveRankingPosition rank
             ranking.CheckpointIndex + 1, ranking.IsFinish);
     }
 
-    private Task<bool> CheckIsRoundsModeAsync()
+    private async Task CheckAndSetRoundsMode()
     {
         List<string> validModes = new List<string>
         {
             "Trackmania/TM_Rounds_Online.Script.txt", "Trackmania/TM_Cup_Online.Script.txt"
         };
-        //TODO: https://github.com/EvoEsports/EvoSC-sharp/issues/234 
-        _isRoundsMode = true;
-        return Task.FromResult(_isRoundsMode);
+        var scriptInfo = await client.Remote.GetModeScriptInfoAsync();
+        if (validModes.Contains(scriptInfo.Name))
+        {
+            _isRoundsMode = true;
+        }
+        _isRoundsMode = false;
```

Comment:
Shouldn't this be an else? Otherwise it always sets it to false. Might be better to simply assign the result of `validModes.Contains(scriptInfo.Name)` to `_isRoundsMode` directly instead anyways.